tscached/cache_calls.py

```diff
@@ -66,11 +67,16 @@ def cold(config, redis_client, kquery, kairos_time_range):
         pipeline.set(mts.get_key(), json.dumps(mts.result), ex=mts.expiry)
         response_kquery = mts.build_response(kairos_time_range, response_kquery, trim=False)
 
-    result = pipeline.execute()
-    success_count = len(filter(lambda x: x is True, result))
-    logging.info("MTS write pipeline: %d of %d successful" % (success_count, len(result)))
+    try:
+        result = pipeline.execute()
+        success_count = len(filter(lambda x: x is True, result))
+        logging.info("MTS write pipeline: %d of %d successful" % (success_count, len(result)))
+
+        kquery.upsert(start_time, end_time)  # TODO
+    except redis.exceptions.RedisError as e:
```

Comment:
It's a very slim edge-case but would you want to also eat errors of upserts in a warm hit? Ofc most likely the redis error would come when you first tried to grab the cached data, but I suppose it is possible for the failure to occur between the initial grab and the merged MTS upsert.